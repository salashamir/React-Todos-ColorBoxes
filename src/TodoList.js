import { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
  const [todos, setTodos] = useState([]);

  const addTodo = (todo) => {
    setTodos((todos) => [...todos, todo]);
  };

  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map((todo) => (
          <Todo
            key={todo.id}
            id={todo.id}
            todoText={todo.todoText}
            handleRemove={removeTodo}
          />
        ))}
      </ul>
    </div>
  );
};

export default TodoList;
