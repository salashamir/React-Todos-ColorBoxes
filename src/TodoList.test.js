import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("should render without crashing", () => {
  render(<TodoList />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new todo", () => {
  const { queryByText, getByLabelText } = render(<TodoList />);
  const input = getByLabelText("New todo:");
  const button = queryByText("Add new todo");

  expect(queryByText("Go to the movies")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "Go to the movies" } });
  fireEvent.click(button);
  expect(queryByText("Go to the movies")).toBeInTheDocument();
});
