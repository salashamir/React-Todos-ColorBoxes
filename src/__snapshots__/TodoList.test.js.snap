// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="todoText"
          >
            New todo:
          </label>
          <input
            id="todoText"
            name="todoText"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Add new todo
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
